fs.go

```diff
@@ -208,3 +210,104 @@ func CopyFile(src string, dest string) error {
 
 	return os.Chmod(dest, srcinfo.Mode())
 }
+
+// hasFilepathPrefix will determine if "path" starts with "prefix" from
+// the point of view of a filesystem.
+//
+// Unlike strings.HasPrefix, this function is path-aware, meaning that
+// it knows that two directories /foo and /foobar are not the same
+// thing, and therefore hasFilepathPrefix("/foobar", "/foo") will return
+// false.
+func hasFilepathPrefix(path, prefix string) bool {
+	if filepath.VolumeName(path) != filepath.VolumeName(prefix) {
+		return false
+	}
+
+	var dn string
+
+	if isDir, err := IsDir(path); err != nil {
+		return false
+	} else if isDir {
+		dn = path
+	} else {
+		dn = filepath.Dir(path)
+	}
+
+	dn = strings.TrimSuffix(dn, string(os.PathSeparator))
+	prefix = strings.TrimSuffix(prefix, string(os.PathSeparator))
+
+	dirs := strings.Split(dn, string(os.PathSeparator))[1:]
+	prefixes := strings.Split(prefix, string(os.PathSeparator))[1:]
+
+	if len(prefixes) > len(dirs) {
+		return false
+	}
+
+	var d, p string
+
+	for i := range prefixes {
+		// need to test each component of the path for
+		// case-sensitiveness because on Unix we could have
+		// something like ext4 filesystem mounted on FAT
+		// mountpoint, mounted on ext4 filesystem, i.e. the
+		// problematic filesystem is not the last one.
+		if isCaseSensitiveFilesystem(filepath.Join(d, dirs[i])) {
+			d = filepath.Join(d, dirs[i])
+			p = filepath.Join(p, prefixes[i])
+		} else {
+			d = filepath.Join(d, strings.ToLower(dirs[i]))
+			p = filepath.Join(p, strings.ToLower(prefixes[i]))
+		}
+
+		if p != d {
+			return false
+		}
+	}
+
+	return true
+}
+
+// flipCase returns an string with all the runes case-flipped
+func flipCase(str string) string {
+	return strings.Map(func(r rune) rune {
```

Comment:
For strings with 128 runes, flipping the first one vs flipping all cuts running time in about half (ToLower, ToUpper are modestly expensive operations), so I'm flipping the first one only as you suggest.